src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package service.impl;
+
+import database.StorageDealer;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final StorageDealer storageDealer;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, StorageDealer storageDealer) {
+        this.operationStrategy = operationStrategy;
+        this.storageDealer = storageDealer;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        Map<String, Integer> toStorage = fruitTransactions.stream()
+                .collect(Collectors.toMap(
+                        FruitTransaction::getFruit,
+                        this::getQuantityOfTransaction,
+                        Integer::sum
+                ));
+        storageDealer.checkBalance(toStorage);
+        storageDealer.updateDatabase(toStorage);
+    }
```

Comment:
It's not clear why you're checking the balance before updating the database. If the balance is negative, you should throw a RuntimeException according to the task description. But in your code, you're just checking the balance and then updating the database regardless of the balance. It seems like you're not handling the case when the balance is negative.